docs/monitoring/autodiscovery.md

```diff
@@ -0,0 +1,60 @@
+
+In container environments, Sematext Agent will create a monitoring agent if it finds `MONITORING_TOKEN` in environment variables or labels of your service or if you enabled automatic monitoring for specific service type in the UI. In Kubernetes setups `MONITORING_TOKEN` can be placed in pod annotations.
+
+Example: Steps required to monitor a containerized application like Elasticsearch manually: 
+1. Visit Sematext Cloud UI, create a new Elasticsearch App, open container env instructions
+2. Following the UI instructions, deploy Sematext Agent container (as a standalone container, as Kubernetes DaemonSet or as global Swarm Service)
+3. Add the Elasticsearch `MONITORING_TOKEN` as label or environment variable to your Elasticsearch container
+4. Restart your Elasticsearch container to apply the label/environment variable
+
+Example: Steps required to monitor a containerized application like Elasticsearch automatically: 
+1. Visit Discovery in UI - open instructions to install Sematext Agent in container env
+2. Following the UI instructions, deploy Sematext Agent container (as a standalone container, as Kubernetes DaemonSet or as global Swarm Service)
+3. From Discovery UI, click on any of discovered services and enable automatic monitoring option
+
+As soon as Sematext Agent discovers the Elasticsearch container having a `MONITORING_TOKEN` set or automatic monitoring option enabled, it will set up monitoring agent container which will start to collect the Elasticsearch metrics. 
+
+### Can I have a mix of manually and automatically monitored services?
+
+Yes. In case of bare-metal/virtual machine environments, just follow the classic installation instructions to set up "manually" monitored service agent. Sematext Agent will recognize it as manually set up and will not apply any of automatic monitoring rules on it. Automatic monitoring can be specified on service type level in Discovery UI and will be applied for any instance of that service except those that already have manually set up agent.
+
+In container envs just add the `MONITORING_TOKEN` environment variable (or label) to container of the service you wish to monitor manually. Automatic monitoring can again be enabled in UI and Sematext Agent will again "know" which service instances have manually set up agent and which should get automatic monitoring rule applied.  
+
+In all environments this makes it possible to use different monitoring tokens, and thus different Sematext Apps for the same service type running on the same machine or in the same Kubernetes/Swarm cluster.
+
```

Comment:
We may need this:
Does Autodiscovery work with <insert a few top managed k8s services, like AWS EKS>?